engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Builtins.java

```diff
@@ -149,13 +159,12 @@ private void registerBuiltinMethods(ModuleScope scope, Language language) {
     for (Builtin builtin : builtins.values()) {
       var type = builtin.getType();
       String tpeName = type.getName();
-      Map<String, Class<BuiltinRootNode>> methods = builtinMethodNodes.get(tpeName);
+      Map<String, Method> methods = builtinMethodNodes.get(tpeName);
       if (methods != null) {
         methods.forEach(
-            (methodName, clazz) -> {
+            (methodName, meth) -> {
               Optional<Function> fun;
               try {
-                Method meth = clazz.getMethod("makeFunction", Language.class);
                 fun = Optional.ofNullable((Function) meth.invoke(null, language));
```

Comment:
Does this really work without any configuration? Is `native-image` clever enough to register existing `Method`instances in the heap for reflection?

Reply:
Yeap, it does work.

Reply:
None that Graal treats `getMethod` a bit specially. Full list is available at https://www.graalvm.org/22.0/reference-manual/native-image/Reflection/